.github/workflows/python_unit_test.yaml

```diff
@@ -28,22 +41,25 @@ jobs:
           --health-interval 10s
           --health-timeout 5s
           --health-retries 5
-    
+
     steps:
       - name: Check out code
         uses: actions/checkout@v3
+        with:
+          persist-credentials: false
       - name: Set up python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.6.7"
+          python-version: ${{ inputs.python-version }}
```

Comment:
we may want to test multiple Python versions but only require that certain versions pass. That would be a really nice to have feature so that we can see if what we're writing is future-compatible (e.g. test on 3.9, 3.10, 3.11 etc but only fail if 3.9 fails).

Reply:
I think we can achieve this by setting up this workflow multiple times, for each version of python we need to test, and setting which ones are required to pass in the repo settings